homeassistant/components/template/lock.py

```diff
@@ -66,7 +115,22 @@ async def async_setup_platform(
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the template lock."""
-    async_add_entities(await _async_create_entities(hass, config))
+    if discovery_info is None:
+        async_add_entities(
+            await _async_create_entities(
+                hass,
+                [rewrite_legacy_to_modern_conf(hass, config)],
+            )
+        )
+        return
+
+    async_add_entities(
+        await _async_create_entities(
+            hass,
+            discovery_info["entities"],
+            discovery_info["unique_id"],
+        )
+    )
```

Comment:
Same comment as for the fan PR: Add a check here rejecting attempts to configure trigger based template fan